rudder-core/src/main/scala/com/normation/rudder/reports/ComplianceMode.scala

```diff
@@ -44,23 +44,66 @@ import net.liftweb.json._
  * - error_only: only report for repaired and error reports.
  */
 
-sealed trait ComplianceMode {
-  def name: String
+sealed trait ComplianceModeName {
+  val name : String
 }
 
-final case object FullCompliance extends ComplianceMode {
+final case object FullCompliance extends ComplianceModeName {
   val name = "full-compliance"
 }
-final case class ChangesOnly (
-  heartbeatPeriod : Int
-)extends ComplianceMode {
-  val name = ChangesOnly.name
-}
 
-object ChangesOnly {
+final case object ChangesOnly extends ComplianceModeName {
   val name = "changes-only"
 }
 
+final case object ReportDisabled extends ComplianceModeName {
+  val name = "report-disabled"
```

Comment:
I was tricked because I looked into the first email about the issue which states 'report-diabled' ;)